src/lsptoolshost/optionNameConverter.ts

```diff
@@ -29,7 +29,13 @@ export function convertServerOptionNameToClientConfigurationName(section: string
         const prefixes = ['dotnet', 'csharp'];
         const optionNamePrefix = getPrefix(optionName, prefixes);
 
-        const featureName = optionNamePrefix == '' ? optionName : optionName.substring(optionNamePrefix.length + 1);
+        let featureName = optionNamePrefix == '' ? optionName: optionName.substring(optionNamePrefix.length + 1);
+        
+        // Handle special case for background analysis option: we want to avoid
+        // 'solution crawler' terminology in the user facing VSCode option.
+        if (featureName =="solution_crawler_background_analysis_scope") {
+            featureName = "analyzer_diagnostics_scope";
+        }
```

Comment:
I added this special handling for mapping the name for this option, but if we feel it is fine to rename the server option name, I can do so as well. I am concerned about this leading to breaking VS users though..